### PR TITLE
Implement `SerializeBytes` for `TlsByteVecUX` types

### PR DESCRIPTION
**Pull request by imor**
This PR implements the `SerializeBytes` trait for all the `TlsByteVecUX` types. It also adds tests for serializing them using the `SerializeBytes` trait.

### CODE REVIEW

File: tls_codec/src/tls_vec.rs [hunk 1]
```diff
@@ -78,6 +78,45 @@ macro_rules! impl_byte_deserialize {
             let result = Self { vec: vec.to_vec() };
             Ok((result, &remainder.get(len..).ok_or(Error::EndOfStream)?))
         }
+
+        fn serialize_bytes_bytes(&$self) -> Result<Vec<u8>, Error> {
```
Comment (franziskuskiefer) [inline tls_codec/src/tls_vec.rs:82]: This looks like a lot of duplicate code. Can you factor out some of it to reuse as much as possible?

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks!
Only one nit.

File: tls_codec/src/tls_vec.rs (hunk 1 above)
Comment (imor) [reply tls_codec/src/tls_vec.rs]: Done.

**Comment by imor** [comment]
Bump, @franziskuskiefer could you please review the changes.

**Comment by imor** [comment]
@franziskuskiefer the changes you requested have been done, could you please review this PR.

File: tls_codec/src/tls_vec.rs [hunk 2]
```diff
@@ -171,6 +149,22 @@ macro_rules! impl_byte_serialize {
...
+
+            // Now serialize the elements
+            written += writer.write($self.as_slice())?;
+
+            $self.assert_written_bytes(tls_serialized_len, written)?;
+            Ok(written)
+        }
+    };
+}
+macro_rules! impl_serialize_common {
```
Comment (franziskuskiefer) [inline tls_codec/src/tls_vec.rs:164]:
```suggestion

macro_rules! impl_serialize_common {
```

File: tls_codec/src/tls_vec.rs [hunk 3]
```diff
@@ -127,38 +127,16 @@ macro_rules! impl_serialize {
...
                 written += e.tls_serialize(writer)?;
             }
 
-            debug_assert_eq!(
-                written, tls_serialized_len,
-                "{} bytes should have been serialized but {} were written",
-                tls_serialized_len, written
-            );
-            if written != tls_serialized_len {
-                return Err(Error::EncodingError(format!(
```
Comment (franziskuskiefer) [inline tls_codec/src/tls_vec.rs:157]: Why did you remove this check in `assert_written_bytes`?

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Generally looks good to me but I think we lost the `EncodingError`?

File: tls_codec/src/tls_vec.rs (hunk 3 above)
Comment (imor) [reply tls_codec/src/tls_vec.rs:157]: This check is present in the `assert_written_bytes` function: see [line 191 in the new file](https://github.com/RustCrypto/formats/blob/a8cd76e4ecc203f176af0acbeca490f560a67352/tls_codec/src/tls_vec.rs#L191).

**Review by franziskuskiefer — APPROVED** [review]
Thanks! Sorry for the delay.
